### DISCUSSION TITLE
Pointing `brew` to local checkout

### DISCUSSION THREAD

**Discussion by mxr** [Homebrew/discussions#4616] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.25
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2ccdb26c45e0ac8c7d4a8c2cc78cd577c28a2707
Last commit: 7 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ee0e263466d20c9be262009c1a7414e96fc4bc05
Core tap last commit: 5 months ago
Core tap branch: master
Core tap JSON: 03 Jul 01:05 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 13.0.0 build 1300
Git: 2.39.1 => /usr/local/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 11.7.7-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  node@12
  python@3.7

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  openssl@3
  tcl-tk

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

I'm (hoping to be) a first-time contributor to `brew`. As part of the PR checklist I'm running `brew style` however from the output I can tell it's executing on the files in the `$(brew --prefix)` directory, not my fork.

How can I ensure `brew style` runs on the code in my fork?

**Comment by SMillerDev** [comment]
Use the code in `$(brew --prefix)`  instead of somewhere else. Everything in Homebrew is geared towards that workflow.

**Comment by osalbahr** [comment, marked as answer]
> How can I ensure brew style runs on the code in my fork?

I’m assuming you forked one of Homebrew’s repos, applied changes, and want the local Homebrew be the fork. I’m not sure how to do that. I recommend creating a branch locally, apply the changes, test, comit, and then push to the fork and PR. See the instructions in [Homebrew Documentation](https://docs.brew.sh/), specifically [How to Open a Pull Request (and get it merged)](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request). If you’re editing a formula, using `brew edit <formula>` is the easiest way to do it (also in the docs).

**Comment by mxr** [comment]
Thanks for clarifying. I will look into sending a PR to the linked docs so it's clearer that brew development should happen in `brew --prefix` for the various lint commands to work
